src/sidebar/store/modules/test/groups-test.js

```diff
@@ -108,4 +139,135 @@ describe('sidebar.store.modules.groups', () => {
+        isLoggedIn: false,
+        filteredGroups: [openGroup],
+        expectedGroups: [openGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups, expectedGroups }) => {
+      it(description, function() {
+        store.getState().session = { userid: isLoggedIn ? '1234' : null };
+        store.loadGroups(filteredGroups);
+
+        const featuredGroups = store.getFeaturedGroups();
+        const currentlyViewing = store.getCurrentlyViewingGroups();
+
+        assert.deepEqual(featuredGroups, expectedGroups);
+        assert.isTrue(currentlyViewing.length === 0);
+      });
+    });
+  });
+
+  describe('getMyGroups', () => {
+    [
+      {
+        description:
+          'If not logged in and the group is the Public group, do not show it in `My Groups`',
+        isLoggedIn: false,
+        filteredGroups: [publicGroup],
+        expectedGroups: [],
+      },
+      {
+        description:
+          'If logged in and the user is a member of the group, show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [openGroup],
+        expectedGroups: [],
+      },
+      {
+        description:
+          'If logged in and the user is a member of the unscoped group, show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [restrictedOutOfScopeMemberGroup],
+        expectedGroups: [restrictedOutOfScopeMemberGroup],
+      },
+      {
+        description:
+          'If logged in and the user is a member of the group, show it in `My Groups`',
+        isLoggedIn: true,
+        filteredGroups: [publicGroup],
+        expectedGroups: [publicGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups, expectedGroups }) => {
+      it(description, function() {
+        store.getState().session = { userid: isLoggedIn ? '1234' : null };
+        store.loadGroups(filteredGroups);
+
+        const myGroups = store.getMyGroups();
+        const currentlyViewing = store.getCurrentlyViewingGroups();
+
+        assert.deepEqual(myGroups, expectedGroups);
+        assert.isTrue(currentlyViewing.length === 0);
+      });
+    });
+  });
+
+  describe('getCurrentlyViewingGroups', () => {
+    [
+      {
+        description:
+          'If logged out and there is an out-of-scope restricted group, show it in `Currently Viewing`',
+        isLoggedIn: false,
+        filteredGroups: [restrictedOutOfScopeGroup],
+      },
+      {
+        description:
+          'If logged out and only the Public group is present, show it in `Currently Viewing`',
+        isLoggedIn: false,
+        filteredGroups: [publicGroup],
+      },
+      {
+        description:
+          'If logged in and there is an out-of-scope restricted group that the user is not a memberof, show it in `Currently Viewing`',
+        isLoggedIn: true,
+        filteredGroups: [restrictedOutOfScopeGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups }) => {
+      it(description, function() {
```

Comment:
Nit: This should use an arrow function. `function () { ... }` should only be used in new code if you are trying to create a method on an object that uses `this`.